backend/app/controllers/spree/admin/promotions_controller.rb

```
@@ -8,24 +8,35 @@ class PromotionsController < ResourceController
 
       helper 'spree/promotion_rules'
 
-      protected
-        def build_promotion_codes
-          @bulk_base = params[:bulk_base] if params[:bulk_base].present?
-          @bulk_number = Integer(params[:bulk_number]) if params[:bulk_number].present?
-
-          if @bulk_base && @bulk_number
-            @promotion.build_promotion_codes(
-              base_code: @bulk_base,
-              number_of_codes: @bulk_number,
-            )
-          end
+      def create
+        @bulk_base = params[:bulk_base].presence
+        @bulk_number = Integer(params[:bulk_number]) if params[:bulk_number].present?
```

Since this is at the admin controller level I'd just making the input field be `type="number"` and let it raise in the unlikely event that someone manages to mess that up, but I wouldn't object to rescuing the `ArgumentError` specifically and messaging that.  But either way I'd vote to use `Integer()` to avoid unexpected surprises from silent `to_i` conversions.